src/java/org/apache/cassandra/db/rows/Cell.java

```diff
@@ -59,14 +69,24 @@
 
     public interface Factory<V>
     {
-        Cell<V> create(ColumnMetadata column, long timestamp, int ttl, int localDeletionTime, V value, CellPath path);
+        Cell<V> create(ColumnMetadata column, long timestamp, int ttl, long localDeletionTime, V value, CellPath path);
     }
 
     protected Cell(ColumnMetadata column)
     {
         super(column);
     }
 
+    public static int deletionTimeLongToGuavaUint(long deletionTime)
+    {
+        return deletionTime == NO_DELETION_TIME ? NO_DELETION_TIME_GUAVA_UINT : UnsignedInts.checkedCast(deletionTime);
```

Comment:
Directly using `UnsignedInts` instead of instantiating `UnsignedInteger` is nice. The `UnsignedInts` class is marked as beta and it shows a distracting warning on the four classes where it's used. The warning seems quite ignorable since the two only methods we use look quite simple and safe.

Perhaps we could avoid the warning by wrapping those two methods into our utility class. That should hide the warning and isolate the Guava dependency. It would also make easy to change the implementation in the future, if we wanted to do so. That wrapping would look like:
```java
package org.apache.cassandra.utils;

import com.google.common.primitives.UnsignedInts;

public class UnsignedIntegers
{
    public static int fromLong(long value)
    {
        return UnsignedInts.checkedCast(value);
    }

    public static long toLong(int value) {
        return UnsignedInts.toLong(value);
    }
}
```
Since we would be encapsulating the Guava dependency into the utility class, we could rename some vars to make them independent of the implementation. For example, `deletionTimeGuavaUint` could be named just `deletionTimeUint`, or perhaps `deletionTimeUnsignedInteger`.

Wdyt?

Reply:
I like this a lot. I've gone a bit further to make everything implementation agnostic. Wdyt?

Reply:
Looks great, +1